In [1]:
# Função que gera o intervalo de confiança
GenInterval <- function(sample, size){
    thetamv <- mean(sample)
    c1 <- -1.96
    c2 <- 1.96
    return(c(-(c2*sqrt(thetamv))/sqrt(size) + thetamv, (-(c1*sqrt(thetamv))/sqrt(size) + thetamv)))
}

In [4]:
# Número de ensaios
trials = 100000
theta = 0.01

# Tamanhos amostrais a serem testados
sample_sizes = seq(1000, 5000, 500)

# Para todo tamanho amostral
for(sample_size in sample_sizes){
    hit = 0
    theta_mv = c()
    # Para cada ensaio
    for(i in 1:trials){
        # Gera amostra
        sample = rpois(sample_size, theta)
        # Armazena EMV
        theta_mv = c(theta_mv,mean(sample))
        # Gera intervalo de confiança
        interval = GenInterval(sample, sample_size)
        # Verifica se theta está no intervalo
        if(interval[1] < theta & interval[2] > theta){
            hit = hit + 1
        }
    }
    # Imprime porcentagem
    cat("Size =", sample_size, " - Percentage =", hit/trials, "\n")
}



Size = 1000  - Percentage = 0.9273 
Size = 1500  - Percentage = 0.92009 
Size = 2000  - Percentage = 0.94671 
Size = 2500  - Percentage = 0.94906 
Size = 3000  - Percentage = 0.93028 
Size = 3500  - Percentage = 0.93641 
Size = 4000  - Percentage = 0.94182 
Size = 4500  - Percentage = 0.94798 
Size = 5000  - Percentage = 0.94933 


Notamos que para $n > 4000$, temos a porcentagem de intervalos que contém $\theta$ mais do que $94\%$ das vezes. Logo, propomos tamanho amostral $n > 4000$. 